packages/x-components/src/components/suggestions/__tests__/base-suggestions.spec.ts

```diff
@@ -32,4 +33,25 @@ describe('testing Base Suggestions component', () => {
       expect(liWrappers.at(index).element.textContent).toEqual(`${index} - ${suggestion.query}`)
     );
   });
+
+  it('renders at most the number of suggestions defined by `maxItemsToRender` prop', async () => {
+    const wrapper = mount(BaseSuggestions, {
+      propsData: { suggestions }
+    });
+    const renderedSuggestions = (): WrapperArray<Vue> =>
+      findTestDataById(wrapper, 'suggestion-item');
+
+    await wrapper.setProps({ maxItemsToRender: 2 });
+    expect(renderedSuggestions()).toHaveLength(2);
+
+    await wrapper.setProps({ maxItemsToRender: 3 });
+    expect(renderedSuggestions()).toHaveLength(3);
+
+    await wrapper.setProps({ maxItemsToRender: 5 });
+    expect(renderedSuggestions()).toHaveLength(suggestions.length);
```

Comment:
You have no idea how much I want to redo this tests with the `renderX` pattern 🤣, but for now it is fine. I would only recommend changing the test to be explicit about what we are testing in each step

If we don't know how many suggestions we have (because we are using an unknown stub) we can always parametrise the values. Note that I added a new assertion to check that by default all suggestions are rendered.

```suggestion
    expect(renderedSuggestions()).toHaveLength(suggestions.length);

    await wrapper.setProps({ maxItemsToRender: suggestions.length - 1 });
    expect(renderedSuggestions()).toHaveLength(suggestions.length - 1);

    await wrapper.setProps({ maxItemsToRender: suggestions.length });
    expect(renderedSuggestions()).toHaveLength(suggestions.length);

    await wrapper.setProps({ maxItemsToRender: suggestions.length + 1 });
    expect(renderedSuggestions()).toHaveLength(suggestions.length);
```